### PR TITLE
net/aria2: update to version 1.37.0

### PR DESCRIPTION
**Pull request by egorenar**
This patch fixes a format string overflow in Logger.cc.

```
 root@OpenWrt:~# cat test.c

 #include <time.h>
 #include <sys/time.h>

 int main(int argc, char** argv)
 {
         printf("sizeof time_t = %d\n", sizeof(time_t));
         printf("sizeof size_t = %d\n", sizeof(size_t));
         printf("sizeof struct timeval = %d\n", sizeof(struct timeval));
         return 0;
 }

 root@OpenWrt:~# gcc test.c

 root@OpenWrt:~# ./a.out

 sizeof time_t = 8
 sizeof size_t = 4
 sizeof struct timeval = 16
```

Maintainer: @kaloz
Compile tested: (Netgear XR700 ARMv7 Processor rev 4 (v7l) OpenWrt SNAPSHOT r24099-5dfd8dc7d4f5)
Run tested: (ARMv7 Processor rev 4 (v7l) ARMv7 Processor rev 4 (v7l) OpenWrt SNAPSHOT r24099-5dfd8dc7d4f5)

### CODE REVIEW

File: net/aria2/patches/0001-Logger-Fix-format-string-overflow-in-writeHeader.patch [hunk 1]
```diff
@@ -0,0 +1,94 @@
+From 8718757e6c62aeffd81a2ed9fa88e0c05b6412cf Mon Sep 17 00:00:00 2001
+From: Alexander Egorenkov <egorenar-dev@posteo.net>
+Date: Sat, 7 Oct 2023 13:39:06 +0200
+Subject: [PATCH 1/1] Logger: Fix format string overflow in writeHeader()
+
+This problem occurred on a 32-bit ARM OpenWRT router running Linux 6.1.
```
Comment (BKPepe) [inline net/aria2/patches/0001-Logger-Fix-format-string-overflow-in-writeHeader.patch:6]: Nitpick: It's OpenWrt.

File: net/aria2/patches/0001-Logger-Fix-format-string-overflow-in-writeHeader.patch [hunk 2]
```diff
@@ -0,0 +1,94 @@
+From 8718757e6c62aeffd81a2ed9fa88e0c05b6412cf Mon Sep 17 00:00:00 2001
+From: Alexander Egorenkov <egorenar-dev@posteo.net>
+Date: Sat, 7 Oct 2023 13:39:06 +0200
+Subject: [PATCH 1/1] Logger: Fix format string overflow in writeHeader()
```
Comment (BKPepe) [inline net/aria2/patches/0001-Logger-Fix-format-string-overflow-in-writeHeader.patch:4]: This patch needs to be upstreamed. I did not see that it was send to upstream.

**Review by BKPepe — CHANGES_REQUESTED** [review]

File: net/aria2/patches/0001-Logger-Fix-format-string-overflow-in-writeHeader.patch (hunk 2 above)
Comment (egorenar) [reply net/aria2/patches/0001-Logger-Fix-format-string-overflow-in-writeHeader.patch:4]: i will give it a try but it seems to be dead :/

File: net/aria2/patches/0001-Logger-Fix-format-string-overflow-in-writeHeader.patch (hunk 1 above)
Comment (egorenar) [reply net/aria2/patches/0001-Logger-Fix-format-string-overflow-in-writeHeader.patch:6]: thx, will fix.

File: net/aria2/patches/0001-Logger-Fix-format-string-overflow-in-writeHeader.patch (hunk 2 above)
Comment (BKPepe) [reply net/aria2/patches/0001-Logger-Fix-format-string-overflow-in-writeHeader.patch:4]: What? The project is not dead. See: https://github.com/aria2/aria2/commits/master

Comment (egorenar) [reply net/aria2/patches/0001-Logger-Fix-format-string-overflow-in-writeHeader.patch:4]: hmm, ok, then i was confused by so many issues and pull requests w/o answer.

**Comment by BKPepe** [comment]
Putting this on hold until the patch is sent to upstream.

**Comment by egorenar** [comment]
> Putting this on hold until the patch is sent to upstream.

https://github.com/aria2/aria2/pull/2117

**Comment by egorenar** [comment]
the fix has been merged and pending for release 1.37.

**Comment by egorenar** [comment]
1.37 released today.
i updated the PR by bumping the version.

**Comment by zdz2019** [comment]
aria2 1.37 it's broken.

see log:

Sun Nov 26 09:41:05 2023 daemon.info aria2: Instance "main" disabled.
Sun Nov 26 09:41:25 2023 daemon.info aria2: Aria2 will run with user 'aria2'.
Sun Nov 26 09:41:25 2023 daemon.info aria2c[6854]: jail: exec-ing /usr/bin/aria2c
Sun Nov 26 09:41:26 2023 daemon.err aria2c[6854]: Exception caught
Sun Nov 26 09:41:26 2023 daemon.err aria2c[6854]: Exception: [Platform.cc:125] errorCode=1 OSSL_PROVIDER_load 'legacy' failed.
Sun Nov 26 09:41:26 2023 daemon.err aria2c[6854]:
Sun Nov 26 09:41:26 2023 daemon.info aria2c[6854]: jail: jail (6856) exited with exit: 1
Sun Nov 26 09:41:31 2023 daemon.info aria2c[6888]: jail: exec-ing /usr/bin/aria2c
Sun Nov 26 09:41:31 2023 daemon.err aria2c[6888]: Exception caught
Sun Nov 26 09:41:31 2023 daemon.err aria2c[6888]: Exception: [Platform.cc:125] errorCode=1 OSSL_PROVIDER_load 'legacy' failed.
Sun Nov 26 09:41:31 2023 daemon.err aria2c[6888]:
Sun Nov 26 09:41:31 2023 daemon.info aria2c[6888]: jail: jail (6889) exited with exit: 1
Sun Nov 26 09:41:36 2023 daemon.info aria2c[6896]: jail: exec-ing /usr/bin/aria2c
Sun Nov 26 09:41:36 2023 daemon.err aria2c[6896]: Exception caught
Sun Nov 26 09:41:36 2023 daemon.err aria2c[6896]: Exception: [Platform.cc:125] errorCode=1 OSSL_PROVIDER_load 'legacy' failed.
Sun Nov 26 09:41:36 2023 daemon.err aria2c[6896]:
Sun Nov 26 09:41:36 2023 daemon.info aria2c[6896]: jail: jail (6897) exited with exit: 1
Sun Nov 26 09:41:41 2023 daemon.info aria2c[6908]: jail: exec-ing /usr/bin/aria2c
Sun Nov 26 09:41:41 2023 daemon.err aria2c[6908]: Exception caught
Sun Nov 26 09:41:41 2023 daemon.err aria2c[6908]: Exception: [Platform.cc:125] errorCode=1 OSSL_PROVIDER_load 'legacy' failed.

**Comment by egorenar** [comment]
Hmm, for me it works perfectly now.

My config:
```
#
# Aria2 Configuration
#
# CONFIG_ARIA2_OPENSSL is not set
CONFIG_ARIA2_GNUTLS=y
# CONFIG_ARIA2_NOSSL is not set
CONFIG_ARIA2_NETTLE=y
# CONFIG_ARIA2_LIBGCRYPT is not set
# CONFIG_ARIA2_NOCRYPTO is not set
CONFIG_ARIA2_LIBXML2=y
# CONFIG_ARIA2_EXPAT is not set
# CONFIG_ARIA2_NOXML is not set
CONFIG_ARIA2_GMP=y
CONFIG_ARIA2_BITTORRENT=y
CONFIG_ARIA2_METALINK=y
CONFIG_ARIA2_SFTP=y
CONFIG_ARIA2_ASYNC_DNS=y
CONFIG_ARIA2_COOKIE=y
CONFIG_ARIA2_WEBSOCKET=y
# end of Aria2 Configuration

```

**Comment by BKPepe** [comment]
Could you put in the commit description link for changelog?

E.g.
```
Release notes:
https://github.com/aria2/aria2/releases/tag/release-1.37.0
```

Afterwards, I will merge it.

**Comment by egorenar** [comment]
> Could you put in the commit description link for changelog?
> 
> E.g.
> 
> ```
> Release notes:
> https://github.com/aria2/aria2/releases/tag/release-1.37.0
> ```
> 
> Afterwards, I will merge it.

done

**Pull request merged**